cengage/react-magma
# fix(Select / Combobox / Dropdown / DatePicker): Menu closing support with escape key in Modals.

Issue: # [264](https://github.com/cengage/react-magma/issues/264)

## What I did
Added support for components with menus inside a Modal to close with the escape key and retain the active Modal.

## Checklist 
- [NA] I have made corresponding changes to the documentation

## How to test
- Go to Modal in Storybook
- Open "Modal In A Modal" example
- Click "Show Modal"
- Click on the Datepicker calendar icon then press escape
- Verify that the Datepicker closes and doesn't close the modal underneath
- Press escape again and confirm modal closes
- Click "Show Modal" then click "Show Modal 2"
- Click on the different dropdowns and select menus, one at a time, then press escape
- Verify that each menu closes and doesn't close the modal underneath
- Press escape again and confirm the top modal closes
- Press escape again and confirm the underlying modal closes
